src/Poseidon/CLI/Forge.hs

```diff
@@ -127,8 +124,33 @@ runForge (ForgeOptions baseDirs entitiesDirect entitiesFile intersect_ outPath o
                 GenotypeFormatEigenstrat -> writeEigenstrat outG outS outI newEigenstratIndEntries
                 GenotypeFormatPlink -> writePlink outG outS outI newEigenstratIndEntries
         liftIO $ hPutStrLn stderr "Processing SNPs..."
-        runEffect $ eigenstratProd >-> printSNPCopyProgress >-> P.map (selectIndices indices) >-> outConsumer
-        liftIO $ hPutStrLn stderr "Done"
+        
+        -- define main forge pipe including file output.
+        -- The final tee forwards the results to be used in the snpCounting-fold
+        let forgePipe = eigenstratProd >->
+                printSNPCopyProgress >->
+                P.map (selectIndices indices) >->
+                P.tee outConsumer
+
+        let startAcc = V.replicate (length newEigenstratIndEntries) 0 :: V.Vector Int
+        P.fold sumNonMissingSNPs startAcc id forgePipe
+    -- janno (with updated SNP numbers)
+    liftIO $ hPutStrLn stderr "Done"
+    hPutStrLn stderr "Creating .janno file"
+    let jannoRowsWithNewSNPNumbers = zipWith (\x y -> x {jNrAutosomalSNPs = Just y})
+                                             relevantJannoRows
+                                             (V.toList newNrAutosomalSNPs)
+    writeJannoFile (outPath </> outName <.> "janno") jannoRowsWithNewSNPNumbers
+
+sumNonMissingSNPs :: V.Vector Int -> (EigenstratSnpEntry, GenoLine) -> V.Vector Int
+sumNonMissingSNPs accumulator (_, geno) =
+    let nonMissingInt = V.map nonMissingToInt geno
+    in V.zipWith (+) accumulator nonMissingInt
```

Comment:
I've also wondered here in this function whether this may not result in a performance or space issue, since we're literally creating a vector for the garbage collector every single SNP. We need to check whether this makes things much slower... It is easy to use a mutable vector here, simply updating each slot in-space, using `IOVector` and `O.foldM`.